doc/source/data/glossary.rst

```diff
@@ -0,0 +1,140 @@
+.. _datasets_glossary:
+
+=====================
+Ray Datasets Glossary
+=====================
+
+.. glossary::
+
+    Batch format
+        The way batches of data are represented.
+
+        Set ``batch_format`` in methods like :meth:`~ray.data.Dataset.iter_batches` and
+        :meth:`~ray.data.Dataset.map_batches` to specify the batch type.
+
+        .. doctest::
+
+            >>> dataset = ray.data.range_table(10)
+            >>> next(iter(dataset.iter_batches(batch_format="numpy", batch_size=5)))
+            {'value': array([0, 1, 2, 3, 4])}
+            >>> next(iter(dataset.iter_batches(batch_format="pandas", batch_size=5)))
+               value
+            0      0
+            1      1
+            2      2
+            3      3
+            4      4
+
+        To learn more about batch formats, read
+        :ref:`UDF Input Batch Formats <transform_datasets_batch_formats>`.
+
+    Block
+        A processing unit of data. A :class:`~ray.data.Dataset` consists of a list of
+        blocks.
+
+        Under the hood, :term:`Datasets <Datasets (library)>` partition :term:`records <Record>`
+        into a set of distributed data blocks. This allows Datasets to perform operations
+        in parallel.
+
+        Unlike a batch, which is a user-facing object, a block is an internal abstraction.
+
+    Block format
+        The way :term:`blocks <Block>` are represented.
+
+        Blocks are represented as Arrow tables, pandas DataFrames, and Python lists.
```

Comment:
just to be specific, can we add link to "Arrow tables" and "pandas DataFrames"?

https://arrow.apache.org/docs/python/generated/pyarrow.Table.html
https://pandas.pydata.org/docs/reference/api/pandas.DataFrame.html